### PR TITLE
Interface generalization: Vehicle -> Entity{S, D, I}

### PR DESCRIPTION
**Pull request by mattuntergassmair**
just a bunch of interface generalizations to make methods work with all kinds of ids, not just Int

### CODE REVIEW

File: src/feature-extraction/lidar_sensor.jl [hunk 1]
```diff
@@ -28,7 +28,7 @@ function LidarSensor(nbeams::Int;
     LidarSensor(angles, ranges, range_rates, max_range, ConvexPolygon(4))
 end
 nbeams(lidar::LidarSensor) = length(lidar.angles)
-function observe!(lidar::LidarSensor, scene::Scene, roadway::Roadway, vehicle_index::Int)
+function observe!(lidar::LidarSensor, scene::Frame{Entity{VehicleState, D, I}}, roadway::Roadway, vehicle_index::Int) where {D<:Union{VehicleDef, BicycleModel}, I}
```
Comment (MaximeBouton) [inline src/feature-extraction/lidar_sensor.jl]: should be `D<:AbstractAgentDefinition` instead of the union

File: src/states/scenes.jl [hunk 2]
```diff
@@ -18,9 +18,9 @@ Base.show(io::IO, scene::Scene) = print(io, "Scene(with $(length(scene)) cars)")
 
 Converts an entity in Vehicle (it is converting the agent definition only)
 """
-function Base.convert(::Type{Vehicle}, veh::Entity{VehicleState, D, Int64}) where D<:AbstractAgentDefinition
+function Base.convert(::Type{Entity{VehicleState, VehicleDef, I}}, veh::Entity{VehicleState, D, I}) where {D<:Union{VehicleDef, BicycleModel}, I}
```
Comment (MaximeBouton) [inline src/states/scenes.jl]: same here, `D<:AbstractAgentDefinition`

File: src/behaviors/sidewalk_pedestrian_model.jl [hunk 3]
```diff
@@ -103,13 +103,13 @@ function update_approaching(ped::Entity{S, D, I},
...
 
-function update_appraising(ped::Entity{S, VehicleDef, I},
+function update_appraising(ped::Entity{S, D, I},
                            scene::Scene,
                            model::SidewalkPedestrianModel,
                            roadway::Roadway,
                            crosswalk::Lane,
                            sw_origin::Lane,
-                           ) where {S, I}
+                           ) where {S, D<:Union{VehicleDef, BicycleModel}, I}
```
Comment (MaximeBouton) [inline src/behaviors/sidewalk_pedestrian_model.jl]: let's replace `veh.def.class` by `class(veh.def)` and use `D<:AbstractAgentDefinition`

File: src/feature-extraction/lidar_sensor.jl (hunk 1 above)
Comment (MaximeBouton) [reply src/feature-extraction/lidar_sensor.jl]: let's replace the call to `veh.def.length` by `length(veh.def)`

File: src/feature-extraction/neighbors_features.jl [hunk 4]
```diff
@@ -48,11 +48,11 @@ end
 
 abstract type VehicleTargetPoint end
 struct VehicleTargetPointFront <: VehicleTargetPoint end
-get_targetpoint_delta(::VehicleTargetPointFront, veh::Entity{VehicleState, VehicleDef, I}) where I = veh.def.length/2*cos(veh.state.posF.ϕ)
+get_targetpoint_delta(::VehicleTargetPointFront, veh::Entity{VehicleState, D, I}) where {D<:Union{VehicleDef, BicycleModel}, I} = veh.def.length/2*cos(veh.state.posF.ϕ)
```
Comment (MaximeBouton) [inline src/feature-extraction/neighbors_features.jl]: same here

File: src/feature-extraction/lidar_sensor.jl (hunk 1 above)
Comment (mattuntergassmair) [reply src/feature-extraction/lidar_sensor.jl]: ok, replaced all occurrences

File: src/feature-extraction/neighbors_features.jl (hunk 4 above)
Comment (mattuntergassmair) [reply src/feature-extraction/neighbors_features.jl]: done :+1:

**Pull request merged**